Gems/Terrain/Assets/Shaders/Terrain/ClipmapComputeHelpers.azsli

```diff
@@ -38,71 +38,80 @@ float GetDetailClipmapScaleInv(uint clipmapLevel)
+    |      |      |center|      |   assuming the size of the clipmap is always even.
+    |      |      |0,0   |      |
+    |______|______|______|______|
+    |      |      |      |1,1   |
+    |      |      |      |      |
+    |______|______|______|______|
+
+    */
+    int2 distance = int2(pixelPosition) - int2(clipmapCenter);
+    int2 clipmapSize = int2(TerrainSrg::m_clipmapData.m_clipmapSizeUint, TerrainSrg::m_clipmapData.m_clipmapSizeUint);
+    int2 halfClipmapSize = clipmapSize / 2;
+    distance = (distance + halfClipmapSize + clipmapSize) % clipmapSize - halfClipmapSize;
+
+    float2 viewRelativePosition = distance * (maxRenderRadius * 2.0f * clipmapScaleInv / TerrainSrg::m_clipmapData.m_clipmapSizeFloat);
     return TerrainSrg::m_clipmapData.m_currentViewPosition + viewRelativePosition;
 }
 
 float2 GetCurrentWorldPositionFromMacroClipmaps(uint2 pixelPosition, uint clipmapLevel)
 {
-    float2 currentClipmapCenter = GetCurrentMacroClipmapCenter(clipmapLevel);
+    uint2 currentClipmapCenter = GetCurrentMacroClipmapCenter(clipmapLevel);
     float clipmapScaleInv = GetMacroClipmapScaleInv(clipmapLevel);
-    float maxRenderDistance = TerrainSrg::m_clipmapData.m_macroClipmapMaxRenderRadius;
-    return GetWorldPosition(currentClipmapCenter, pixelPosition, clipmapLevel, clipmapScaleInv, maxRenderDistance);
+    float maxRenderRadius = TerrainSrg::m_clipmapData.m_macroClipmapMaxRenderRadius;
+    return GetWorldPosition(currentClipmapCenter, pixelPosition, clipmapLevel, clipmapScaleInv, maxRenderRadius);
 }
 
 float2 GetCurrentWorldPositionFromDetailClipmaps(uint2 pixelPosition, uint clipmapLevel)
 {
-    float2 currentClipmapCenter = GetCurrentDetailClipmapCenter(clipmapLevel);
+    uint2 currentClipmapCenter = GetCurrentDetailClipmapCenter(clipmapLevel);
     float clipmapScaleInv = GetDetailClipmapScaleInv(clipmapLevel);
-    float maxRenderDistance = TerrainSrg::m_clipmapData.m_detailClipmapMaxRenderRadius;
-    return GetWorldPosition(currentClipmapCenter, pixelPosition, clipmapLevel, clipmapScaleInv, maxRenderDistance);
+    float maxRenderRadius = TerrainSrg::m_clipmapData.m_detailClipmapMaxRenderRadius;
+    return GetWorldPosition(currentClipmapCenter, pixelPosition, clipmapLevel, clipmapScaleInv, maxRenderRadius);
 }
 
-float2 ddxPosition(float maxRenderDistance, float clipmapScaleInv)
+float2 ddxPosition(float maxRenderRadius, float clipmapScaleInv)
 {
-    float dx = maxRenderDistance * 2.0 * clipmapScaleInv / TerrainSrg::m_clipmapData.m_clipmapSize;
+    float dx = maxRenderRadius * 2.0 * clipmapScaleInv / TerrainSrg::m_clipmapData.m_clipmapSizeFloat;
     return float2(dx, 0.0);
 }
 
-float2 ddyPosition(float maxRenderDistance, float clipmapScaleInv)
+float2 ddyPosition(float maxRenderRadius, float clipmapScaleInv)
 {
-    float dy = maxRenderDistance * 2.0 * clipmapScaleInv / TerrainSrg::m_clipmapData.m_clipmapSize;
+    float dy = maxRenderRadius * 2.0 * clipmapScaleInv / TerrainSrg::m_clipmapData.m_clipmapSizeFloat;
```

Comment:
You have this calculation:
`maxRenderRadius * 2.0 * clipmapScaleInv / TerrainSrg::m_clipmapData.m_clipmapSizeFloat`
repeated 3 times in this file, would be better to wrap it in a function with a descriptive name and maybe a comment or two about the details of the calculation.